torch_agents/cogment_verse_torch_agents/muzero/adapter.py

```diff
@@ -0,0 +1,806 @@
+# Copyright 2021 AI Redefined Inc. <dev+cogment@ai-r.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from collections import defaultdict, namedtuple
+import ctypes
+import copy
+import time
+
+import logging
+import torch
+import torch.multiprocessing as mp
+from multiprocessing.pool import ThreadPool
+import numpy as np
+import queue
+
+from data_pb2 import (
+    MuZeroTrainingRunConfig,
+    MuZeroTrainingConfig,
+    AgentAction,
+    TrialConfig,
+    ActorParams,
+    EnvironmentConfig,
+    EnvironmentParams,
+    ActorConfig,
+)
+
+from cogment_verse.utils import LRU
+from cogment_verse import AgentAdapter
+from cogment_verse import MlflowExperimentTracker
+from cogment_verse_torch_agents.wrapper import np_array_from_proto_array, proto_array_from_np_array
+from cogment_verse_torch_agents.muzero.replay_buffer import Episode, TrialReplayBuffer, EpisodeBatch
+from cogment_verse_torch_agents.muzero.agent import MuZeroAgent
+
+from cogment.api.common_pb2 import TrialState
+import cogment
+
+
+# pylint: disable=arguments-differ
+
+log = logging.getLogger(__name__)
+
+
+MuZeroSample = namedtuple("MuZeroSample", ["state", "action", "reward", "next_state", "done", "policy", "value"])
+
+
+class LinearScheduleWithWarmup:
```

Comment:
This should be extract to its own file (even if it's still local to the muzero adapter)

Reply:
Done